common/persistence/nosql/nosqlplugin/cassandra/tasks.go

```diff
@@ -89,6 +89,12 @@ func toTaskListPartitionConfig(v interface{}) *persistence.TaskListPartitionConf
 	version := partition["version"].(int64)
 	numRead := partition["num_read_partitions"].(int)
 	numWrite := partition["num_write_partitions"].(int)
+
+	// even if null is stored in cassandra, gocql will return empty struct instead of nil
+	if version == 0 && numRead == 0 && numWrite == 0 {
+		return nil
+	}
+
```

Comment:
I'm kinda assuming this is a change in behavior?  I don't really see it in the changelog, but I might just lack gocql-context / keyword-guessing-ability to find it: https://github.com/apache/cassandra-gocql-driver/blob/trunk/CHANGELOG.md

If this is a broad "was null -> now empty struct" change, we'll probably have to do this kinda carefully :\  and/or wrap gocql and convert it everywhere, if that's possible.  seems like the sort of change that'd cause a lot of minor and hard-to-notice bugs.